### PR TITLE
OSDOCS-4154 Adding AWS Marketplace install content [4.9]

### PR DESCRIPTION
**Pull request by bscott-rh**
[OSDOCS-4154](https://issues.redhat.com//browse/OSDOCS-4154)
4.9 backport for https://issues.redhat.com/browse/OSDOCS-3030

This is a continuation of https://github.com/openshift/openshift-docs/pull/48413

Version(s)
4.9

Doc preview:
[Configuring an AWS account](https://bscott-rh.github.io/openshift-docs/OSDOCS-4154/installing/installing_aws/installing-aws-account.html#installation-aws-marketplace_installing-aws-account)
[Installing on AWS with customizations](https://bscott-rh.github.io/openshift-docs/OSDOCS-4154/installing/installing_aws/installing-aws-customizations.html#installation-aws-marketplace-subscribe_installing-aws-customizations)
[Installing on AWS on government or secret regions](https://bscott-rh.github.io/openshift-docs/OSDOCS-4154/installing/installing_aws/installing-aws-government-region.html#installation-aws-marketplace-subscribe_installing-aws-government-region)
[Creating the worker nodes in AWS](https://bscott-rh.github.io/openshift-docs/OSDOCS-4154/installing/installing_aws/installing-aws-user-infra.html#installation-creating-aws-worker_installing-aws-user-infra)

### CODE REVIEW

**Comment by bscott-rh** [comment]
@yunjiang29 @mjpytlak PTAL at this 4.9 backport for AWS Marketplace install. Thank you

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+endif::[]
+
+:_content-type: PROCEDURE
+[id="installation-aws-marketplace-subscribe_{context}"]
+= Obtaining an AWS Marketplace image
+If you are deploying an {product-title} cluster using an AWS Marketplace image, you must first subscribe through AWS. Subscribing to the offer provides you with the AMI ID that the installation program uses to deploy worker nodes.
+
+[IMPORTANT]
+====
+Deploying an {product-title} cluster using an AWS Marketplace image is not supported in Top Secret regions.
```
Comment (mjpytlak) [inline modules/installation-aws-marketplace-subscribe.adoc:24]:
```suggestion
Deploying an {product-title} cluster using an AWS Marketplace image is not supported in secret regions. 
```
Deploying an {product-title} cluster using an AWS Marketplace image is not supported in secret regions.

File: modules/installation-creating-aws-worker.adoc [hunk 2]
```diff
@@ -180,11 +180,11 @@ the CloudFormation template for the security group and roles.
 section of this topic and save it as a YAML file on your computer. This template
 describes the networking objects and load balancers that your cluster requires.
 
-. If you specified an `m5` instance type as the value for `WorkerInstanceType`,
-add that instance type to the `WorkerInstanceType.AllowedValues` parameter
-in the CloudFormation template.
+. If you specified an `m5` instance type as the value for `WorkerInstanceType`, add that instance type to the `WorkerInstanceType.AllowedValues` parameter in the CloudFormation template.
```
Comment (mjpytlak) [inline modules/installation-creating-aws-worker.adoc:183]:
```suggestion
. Optional: If you specified an `m5` instance type as the value for `WorkerInstanceType`, add that instance type to the `WorkerInstanceType.AllowedValues` parameter in the CloudFormation template.
```

**Review by mjpytlak — COMMENTED** [review]
@bscott-rh Left a couple of suggestions. Also - it looks like you do not update the UPI assembly with the `installation-aws-marketplace-subscribe.adoc`.

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 1 above)
Comment (bscott-rh) [reply modules/installation-aws-marketplace-subscribe.adoc]: Sounds good to me.

File: modules/installation-creating-aws-worker.adoc (hunk 2 above)
Comment (bscott-rh) [reply modules/installation-creating-aws-worker.adoc]: Thanks, updated

**Comment by bscott-rh** [comment]
> @bscott-rh Left a couple of suggestions. Also - it looks like you do not update the UPI assembly with the `installation-aws-marketplace-subscribe.adoc`.

Updated the UPI assembly with the subscribe module. Thank you

File: modules/installation-creating-aws-worker.adoc [hunk 3]
```diff
@@ -180,11 +180,11 @@ the CloudFormation template for the security group and roles.
 section of this topic and save it as a YAML file on your computer. This template
 describes the networking objects and load balancers that your cluster requires.
 
-. If you specified an `m5` instance type as the value for `WorkerInstanceType`,
-add that instance type to the `WorkerInstanceType.AllowedValues` parameter
-in the CloudFormation template.
+. Optional: If you specified an `m5` instance type as the value for `WorkerInstanceType`, add that instance type to the `WorkerInstanceType.AllowedValues` parameter in the CloudFormation template.
```
Comment (yunjiang29) [inline modules/installation-creating-aws-worker.adoc:183]: The allowed values had been removed by https://github.com/openshift/installer/commit/5db41b0a5055c1a98c92f8899c4689ec75cf4bc8, I think we can remove this `Optional` part.

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  platform:
+    aws:
+      amiID: ami-06c4d345f7c207239 <1>
+      type: m5.4xlarge
+  replicas: 3
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-east-2 <2>
```
Comment (yunjiang29) [inline modules/installation-aws-marketplace-subscribe.adoc:60]: Better a real region while in GovCloud, e.g. `us-gov-west-1`

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+endif::[]
+
+:_content-type: PROCEDURE
+[id="installation-aws-marketplace-subscribe_{context}"]
+= Obtaining an AWS Marketplace image
+If you are deploying an {product-title} cluster using an AWS Marketplace image, you must first subscribe through AWS. Subscribing to the offer provides you with the AMI ID that the installation program uses to deploy worker nodes.
+
+[NOTE]
+====
+Deploying an {product-title} cluster using an AWS Marketplace image is not supported in secret regions.
```
Comment (yunjiang29) [inline modules/installation-aws-marketplace-subscribe.adoc:24]: OCP 4.9 is the first version that supports China regions, so adding China regions here?

File: modules/installation-creating-aws-worker.adoc (hunk 3 above)
Comment (yunjiang29) [reply modules/installation-creating-aws-worker.adoc:183]: Oh, I noticed `Allowed values` are still in the document, looks like we need to change lots of things.
So please ignore my above comment, and let's create a separate PR to fix this. @bscott-rh wdyt?

Comment (bscott-rh) [reply modules/installation-creating-aws-worker.adoc:183]: Started a thread in installer Slack to discuss this.

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 4 above)
Comment (bscott-rh) [reply modules/installation-aws-marketplace-subscribe.adoc:60]: Updated doc to reflect this, thanks

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 5 above)
Comment (bscott-rh) [reply modules/installation-aws-marketplace-subscribe.adoc:24]: Updated doc to reflect this, thanks

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  replicas: 3
+metadata:
+  name: test-cluster
+platform:
+  aws:
+ifndef::gov-region[]
+    region: us-east-2 <2>
+endif::gov-region[]
+ifdef::gov-region[]
+    region: gov-us-west-1 <2>
```
Comment (yunjiang29) [inline modules/installation-aws-marketplace-subscribe.adoc:65]: should be `us-gov-west-1`

Comment (bscott-rh) [reply modules/installation-aws-marketplace-subscribe.adoc:65]: Oops, thanks.

**Review by mjpytlak — COMMENTED** [review]
@bscott-rh peer review LGTM.

**Pull request merged**